sweep/sweeper.go

```diff
@@ -1203,13 +1214,29 @@ func (s *UtxoSweeper) mempoolLookup(op wire.OutPoint) fn.Option[wire.MsgTx] {
 	return s.cfg.Mempool.LookupInputMempoolSpend(op)
 }
 
-// handleNewInput processes a new input by registering spend notification and
-// scheduling sweeping for it.
-func (s *UtxoSweeper) handleNewInput(input *sweepInputMessage) error {
+// calculateDefaultDeadline calculates the default deadline height for a sweep
+// request that has no deadline height specified.
+func (s *UtxoSweeper) calculateDefaultDeadline(pi *SweeperInput) int32 {
 	// Create a default deadline height, which will be used when there's no
 	// DeadlineHeight specified for a given input.
 	defaultDeadline := s.currentHeight + int32(s.cfg.NoDeadlineConfTarget)
 
+	// If the input is immature and has a locktime, we'll use the locktime
+	// height as the starting height.
+	matured, locktime := pi.isMature(uint32(s.currentHeight))
+	if !matured {
+		defaultDeadline = int32(locktime + s.cfg.NoDeadlineConfTarget)
```

Comment:
do we need to apply `s.cfg.NoDeadlineConfTarget` to this input? cause isnt that only for inputs with no deadline meaning those will always be mature and wont actually end up here?

Reply:
The deadline height is also the width of the feefunction so we need to set it.

Reply:
yeah suppose the locktime is 40, and the current height is 1000, it means the input cannot be swept until 1040. Then at block height 1040, the input will be swept, and the fee func will need to have a deadline delta, which is default to this config value if not set.